CIP-0062/README.md

```diff
@@ -0,0 +1,220 @@
+---
+CIP: 62
+Title: Cardano dApp-Connector Governance extension
+Authors: Bruno Martins <bruno.martins@iohk.io>
+Status: Draft
+Type: Standards
+Created: 2021-06-11
+License: CC-BY-4.0
+---
+
+# **Abstract**
+
+This document describe the interface between webpage / web-based stack and cardano wallets. This specificies that API of the javascript object that need to be injected into the web applications in order to support all the Governance features.
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# **Motivation**
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# **Specification**
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+
+```
+
+`votingKey`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
+
+```
+
+`Purpose`: Defines the purpose of the delegations. This is used to limit the scope of the delegations.  For example, a purpose might be a subset of Catalyst proposals, a council election, or even some private purpose (agreed by convention).
+
+### **KeyPath**
+```
+interface KeyPath = {
+  address_index: number 
+  account: number
+  role: number
+}
+```
+
+
+
+## **Namespace**
+
+### **cardano.{walletName}.governance.enable(): Promise\<API>**
+The `cardano.{walletName}.governance.enable()` method is used to enable the governance API. It should request permission from the wallet to enable the API. If permission is granted, the rest of the API will be available. The wallet should maintain a specific whitelist of allowed clients for this API. This whitelist can be used to avoid asking for permission every time.
+
+This api being an extension of [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/), expects that `cardano.{walletName}.enable()` to be enabled and added to CIP-30 whitelist implicitly.
+
+When both this API and **CIP-30** is being enabled, is up to the wallet to decide the number of prompts requesting permissions to be displayed to the user.
+
+# **`Jormungandr API`**
+
+## **api.submitVotes**(keyPath: KeyPath, votes: Vote[], spendingCounter: number): Promise\<hash32>
```

Comment:
I would recommend to split that function, or more specifically, to remove responsibilities from that function to keep the CIP-0030 API as orthogonal as possible. There are already is a function for submitting transactions (i.e. `submitTx`). Thus all that is really needed here is to:

- (a) create a transaction "skeleton" that contains the rightful metadata
- (b) sign that transaction with the appropriate signing keys (i.e. the stake keys)

I think both could be done at the same time, and return a partially signed transaction (for which input witnesses may still be provided later by `signTx`). So that the transaction can be inspected before submission.